src/api/Auth/auth.ts

```diff
@@ -0,0 +1,22 @@
+import BaseApi from 'api/BaseApi';
+import TObjectLiteral from 'types/ObjectLiteral';
+
+const AuthApi = {
+    signup: (user: TObjectLiteral) => {
+        const body = JSON.stringify(user);
+
+        return BaseApi.post('auth/signup', body)
+            .then(res => res.json())
+            .then(res => {
+                const { error, reason, id } = res;
+
+                if (id === undefined) {
```

Comment:
```suggestion
                if (typeof id === 'undefined') {
```
Так как-то правильнее